src/extensions/web/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/RazorMappedTextReplacer.cs

```diff
@@ -162,20 +162,61 @@ private static void MinimizeReplacement(ref string original, ref string updated)
                 return;
             }
 
+            // Work at the "word" or token level so that "Hello userA!" -> "Hello userB!" doesn't minimize to replacing all 'A's with 'B's
+            var originalTokens = Tokenize(original).ToArray();
+            var updatedTokens = Tokenize(updated).ToArray();
+
             var index = 0;
-            while (index < original.Length && index < updated.Length && original[index] == updated[index])
+            while (index < originalTokens.Length && index < updatedTokens.Length && originalTokens[index].Equals(updatedTokens[index], StringComparison.Ordinal))
             {
                 index++;
             }
 
             var endIndex = 0;
-            while (endIndex > -original.Length && endIndex > -updated.Length && original[original.Length - 1 + endIndex] == updated[updated.Length - 1 + endIndex])
+            while (endIndex > -originalTokens.Length && endIndex > -updatedTokens.Length && originalTokens[originalTokens.Length - 1 + endIndex].Equals(updatedTokens[updatedTokens.Length - 1 + endIndex], StringComparison.Ordinal))
             {
                 endIndex--;
             }
 
-            original = original.Substring(index, Math.Max(0, original.Length + endIndex - index));
-            updated = updated.Substring(index, Math.Max(0, updated.Length + endIndex - index));
+            original = string.Concat(originalTokens.Skip(index).Take(Math.Max(0, originalTokens.Length + endIndex - index)));
+            updated = string.Concat(updatedTokens.Skip(index).Take(Math.Max(0, updatedTokens.Length + endIndex - index)));
+        }
+
+        /// <summary>
+        /// Splits a string into 'words' where a word is any string of letters and numbers or a single character that isn't a letter or number.
+        /// </summary>
+        /// <param name="str">The string to divide into words.</param>
+        /// <returns>An enumerable of words and all intervening characters comprising the input string.</returns>
+        private static IEnumerable<string> Tokenize(string str)
+        {
```

Comment:
Why create a custom implementation to tokenize words, will `str.Split(new char[] { ' ', '.', '?' }, StringSplitOptions.RemoveEmptyEntries)`  not suffice?

Reply:
`string.Split` doesn't include the characters that split in the output, unfortunately. I need to split "A B" into ["A", " ", "B"], not ["A", "B"]

Reply:
Ah I see, makes sense. The other alternative is to do some regex magic to retain the delimiter too.